### PR TITLE
add kafka exporter

### PR DESCRIPTION
**Pull request by yancl**
@odeke-em  please take a look when you have time, very appreciate.

BTW, if this can be accepted, may i transfer the repo "https://github.com/yancl/opencensus-go-exporter-kafka"  from my personal account to the opencensus group such as census-ecosystem? thanks.

### CODE REVIEW

**Comment by odeke-em** [comment]
Hello @yancl, great to hear back from you and this is great! Nice to see the open sourced Apache Kafka exporter. Let me ping some folks to create a repository and then transfer it there as well as add the Go vanity URL. Thank you very much @yancl.

Also please feel free to add your company with a PR to https://github.com/census-instrumentation/opencensus-website to add it to https://opencensus.io/introduction/ as well as https://opencensus.io/community/users/

**Comment by yancl** [comment]
@odeke-em  very very appreciate for your review, thanks!!

>Also please feel free to add your company with a PR to https://github.com/census-instrumentation/opencensus-website to add it to https://opencensus.io/introduction/ as well as https://opencensus.io/community/users/

I will make a PR when the things related are completed, thanks again, @odeke-em :)

File: exporter/exporterparser/kafka.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterparser
+
+import (
+	"context"
+	"fmt"
```
Comment (odeke-em) [inline exporter/exporterparser/kafka.go]: Please separate stdlib imports from other imports i.e.
```go
import (
    "context"
    "fmt"

    "go.opencensus.io/trace"
    ...
)
```

File: exporter/exporterparser/kafka.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		} `yaml:"exporters"`
+	}
+
+	if err := yamlUnmarshal(config, &cfg); err != nil {
+		return nil, nil, err
+	}
+	if cfg.Exporters == nil {
+		return nil, nil, nil
+	}
+	sc := cfg.Exporters.Kafka
```
Comment (odeke-em) [inline exporter/exporterparser/kafka.go]: kc to mean `kafka config` :)

**Comment by odeke-em** [comment]
I've marked it as "BLOCKED", "DO NOT MERGE" so that we'll merge it after moving the exporter to that location that we've talked about. Thanks again!

File: exporter/exporterparser/kafka.go (hunk 1 above)
Comment (yancl) [reply exporter/exporterparser/kafka.go]: Got it, thanks.

File: exporter/exporterparser/kafka.go (hunk 2 above)
Comment (yancl) [reply exporter/exporterparser/kafka.go]: Yes:)

**Comment by yancl** [comment]
Ok, thanks.

**Comment by flands** [comment]
This is great work and a great addition to the OC project. Given it will be part of OC this means going forward it needs to be supported. I do not see any unit or integration tests for this exporter. This should be addressed prior to merging.

Also, we need an issue to move the repo under OC from github.com/yancl/opencensus-go-exporter-kafka to ensuring ongoing compatibility and support.

**Comment by odeke-em** [comment]
Thanks for taking a look at this @flands!
In regards to
>do not see any unit or integration tests for this exporter. This should be addressed prior to merging.

We don't have those for other exporters as we are just using them like they'd be used in client applications, that is because testing most exporters gets tricky for example Jaeger, Stackdriver, Kafka, Zipkin etc would require setting up servers and adhering to their wire protocols which can be overkill for code that transports data outside and the code also lives elsewhere. For receivers/interceptors, the story is different and those we def require unit and integration tests.

 >Also, we need an issue to move the repo under OC from github.com/yancl/opencensus-go-exporter-kafka to ensuring ongoing compatibility and support.

I don't know where we would open such an issue, as the goal is to eventually add it to https://github.com/census-ecosystem and I don't think that issue should be on this repository. @jparsana and @bogdandrutu are working on creating such a repository but it is a process that takes time.

Having answered those question, I think for now as @songy23 mentioned in https://github.com/census-instrumentation/opencensus-service/pull/142#issuecomment-436706648, @yancl please perform a rebase from master and we can merge it, with the exporter repository living on your own repository and once the repository is finally created in https://github.com/census-ecosystem we can then get in touch about migrating it.

**Comment by yancl** [comment]
@flands thanks for your review, @odeke-em rebase done, PTAL:)

**Comment by yancl** [comment]
Thank you everyone for your great help too:)

**Pull request merged**